simphony/tools/lattice_tools.py

```diff
@@ -0,0 +1,990 @@
+from itertools import permutations, combinations, product
+
+import numpy
+from simphony.cuds.primitive_cell import BravaisLattice
+
+
+def vector_len(vec):
+    ''' Length of vector
+
+    Parameter
+    ---------
+    vec : array_like
+
+    Returns
+    -------
+    length : ndarray
+    '''
+    return numpy.sqrt(numpy.dot(vec, vec))
+
+
+def cosine_two_vectors(vec1, vec2):
+    ''' Return the cosine of the acute angle between two vectors
+
+    Parameters
+    ----------
+    vec1 : array_like
+    vec2 : array_like
+
+    Returns
+    -------
+    cosine : numpy ndarray
+    '''
+    vec1_length = vector_len(vec1)
+    vec2_length = vector_len(vec2)
+    return numpy.dot(vec1, vec2)/vec1_length/vec2_length
+
+
+def same_primitive_cell_config(v1, v2, v3, p1, p2, p3, permute=True,
+                               tolerance=1.e-6):
+    ''' Return True if a set of primitive vectors ``p1``, ``p2``,
+    ``p3`` describe a primitive cell that is geometrically similar to
+    another primitive cell described by ``(v1, v2, v3)``.
+
+    This function works by comparing length ratios and cosines
+    of the angles between vectors.  Single precision applies
+    by default, or change ``tolerance``
+
+    Parameters
+    ----------
+    v1, v2, v3 : array_like (len=3)
+        Primitive vectors of one primitive cell
+    p1, p2, p3 : array_like (len=3)
+        Primitive vectors of another primitive cell
+    permute: boolean
+        whether p1, p2, p3 are permutated
+    tolerance: float
+        tolerance for numerical errors
+
+    Returns
+    -------
+    match : bool
+    '''
+    targets = (v1, v2, v3)
+
+    # cosine angles between pairs of the target primitive vectors
+    target_cosines = numpy.abs([cosine_two_vectors(vec1, vec2)
```

Comment:
Actually, translation invariance is not tested here as there is no comparison of the origin parameters. Primitive vectors themselves, of course, do not hold this information. So, please, ignore that part from my last comment.

Maybe the docstrings could say something along the lines of "Return True if a set of primitive vectors p1, p2, p3 describe a primitive cell that is similar under rotation and/or reflection to the cell described by v1, v2, v3."